File: Inversion/serial/NA/src/hv_subs/PROGRAMS.330/CALPLOT/src/flib/ginfo.f
c---------------------------------------------------------------------c
c                                                                     c
c      COMPUTER PROGRAMS IN SEISMOLOGY                                c
c      VOLUME I                                                       c
c                                                                     c
c      PROGRAM: GINFO                                                 c
c                                                                     c
c      COPYRIGHT (C)  1997 R. B. Herrmann                             c
c                                                                     c
c      Department of Earth and Atmospheric Sciences                   c
c      Saint Louis University                                         c
c      3507 Laclede Avenue                                            c
c      St. Louis, Missouri 63103                                      c
c      U. S. A.                                                       c
c                                                                     c
c---------------------------------------------------------------------c

        subroutine  ginfo(HasMouse, XminDev, YminDev, 
     1      XmaxDev, YmaxDev, XminClip, 
     2      YminClip, XmaxClip, YmaxClip, Color)
        integer*4 HasMouse
        real*4 XminDev, YminDev, XmaxDev, YmaxDev
        real*4 XminClip, YminClip, XmaxClip, YmaxClip
        integer*4 Color

        integer*4  gHasMouse, gXminDev, gYminDev, gXmaxDev, gYmaxDev
        integer*4  gXminClip, gYminClip, gXmaxClip, gYmaxClip
        integer*4 gColor
        common/Xcplot/xold,yold,xcur,ycur
        common/Ycplot/xstp,ystp

        call dfinfo(gHasMouse, gXminDev, gYminDev, 
     1      gXmaxDev, gYmaxDev, gXminClip, 
     2      gYminClip, gXmaxClip, gYmaxClip,gColor)
        HasMouse = gHasMouse 
C       For Device boundaries, use absolute coordinates 
        XminDev = gXminDev 
        YminDev = gYminDev 
        XmaxDev = gXmaxDev 
        YmaxDev = gYmaxDev 
C       return clip coordinates with respect to current origin 
        XminClip = gXminClip /1000.0 
        YminClip = gYminClip /1000.0 
        XmaxClip = gXmaxClip /1000.0 
        YmaxClip = gYmaxClip /1000.0 
        XminClip = (XminClip-xold)/xstp
        YminClip = (YminClip-yold)/ystp
        XmaxClip = (XmaxClip-xold)/xstp
        YmaxClip = (YmaxClip-yold)/ystp
        Color = gColor
        return
        end

File: Inversion/parallel/NA/src/hv_subs/PROGRAMS.330/CALPLOT/src/clib/ginfo.c
/*
c---------------------------------------------------------------------c
c                                                                     c
c      COMPUTER PROGRAMS IN SEISMOLOGY                                c
c      VOLUME I                                                       c
c                                                                     c
c      PROGRAM: GINFO                                                 c
c                                                                     c
c      COPYRIGHT (C)  1997 R. B. Herrmann                             c
c                                                                     c
c      Department of Earth and Atmospheric Sciences                   c
c      Saint Louis University                                         c
c      3507 Laclede Avenue                                            c
c      St. Louis, Missouri 63103                                      c
c      U. S. A.                                                       c
c                                                                     c
c---------------------------------------------------------------------c
*/

#include "callib.h"

#include "dosubs.h"
#include "calplot.h"
 
#include <stdio.h>

extern struct Gcplot Gcplt;

void  ginfo(int *HasMouse, float *XminDev, float *YminDev, 
	float *XmaxDev, float *YmaxDev, float *XminClip, 
	float *YminClip, float *XmaxClip, float *YmaxClip, int *Color)
{
	INT gHasMouse, gXminDev, gYminDev, gXmaxDev, gYmaxDev;
	INT gXminClip, gYminClip, gXmaxClip, gYmaxClip, gColor;
	(*do_info)(&gHasMouse, &gXminDev, &gYminDev, 
		&gXmaxDev, &gYmaxDev, &gXminClip, 
		&gYminClip, &gXmaxClip, &gYmaxClip,&gColor);
	*HasMouse = gHasMouse ;
	/* For Device boundaries, use absolute coordinates */
	*XminDev = gXminDev ;
	*YminDev = gYminDev ;
	*XmaxDev = gXmaxDev ;
	*YmaxDev = gYmaxDev ;
	/* return clip coordinates with respect to current origin */
	*XminClip = gXminClip /1000.0 ;
	*YminClip = gYminClip /1000.0 ;
	*XmaxClip = gXmaxClip /1000.0 ;
	*YmaxClip = gYmaxClip /1000.0 ;
	*XminClip = (*XminClip-Gcplt.xold)/Gcplt.xstp;
	*YminClip = (*YminClip-Gcplt.yold)/Gcplt.ystp;
	*XmaxClip = (*XmaxClip-Gcplt.xold)/Gcplt.xstp;
	*YmaxClip = (*YmaxClip-Gcplt.yold)/Gcplt.ystp;
	*Color = gColor;
}
